### DISCUSSION TITLE
Latest Firefox

### DISCUSSION THREAD

**Discussion by AlphaElwedritsch** [SpiralLinux/SpiralLinux-project#361] [Q&A] [answered]
Hi at all,
@geckolinux what do you recommend to install latest firefox version on SpiralLinux.
I have to use latest instead of ESR.

native/traditional way? Including unstable repo, or using flatpak version?

https://www.debugpoint.com/install-firefox-debian/

Thanks, have a nice day...
Alpha

**Comment by geckolinux** [comment, marked as answer]
Hi there, probably the Flatpak version would be the easiest.

**Comment by AlphaElwedritsch** [comment]
Did it

**Comment by Anonymo** [comment]
https://blog.nightly.mozilla.org/2023/10/30/introducing-mozillas-firefox-nightly-deb-packages-for-debian-based-linux-distributions/

There's an official nightly repo here, perhaps it will move packages to a stable repo in some time.

Also, probably not the best way to do it but getting it from Sid
https://www.jurisic.org/post/2022/10/27/How-to-install-Firefox-from-Debian-Sid-on-Bookworm

**Reply by AlphaElwedritsch** [reply]
ok, thanks
but that are nightlies ... for me to unstable and I don't really want the update frequency every day
and the second suggestion, mmh, i don't really know if i want that way...

so, at the end, i think flatpak is the easiest way. i will keep it that

**Comment by AlphaElwedritsch** [comment]
Firefox has enhanced their native repo to beta versions and in comming versions the finals should also be included.
There's no longer a need for flatpak then.

**Comment by AlphaElwedritsch** [comment]
Firefox has now moved the stable version to their repo, too

I now have a question:

What is more advisable? Integrate a repo directly from the manufacturer or use flatpak?

I have included two repos. Brave and Firefox browsers.
I get other apps that I still need from flatpak.

@geckolinux  what is your opinion? Everything from flatpak and no custom repos? To leave the Debian unchanged?
Or, as in my case, if you know the source exactly, you're more likely to access the repo?

**Reply by geckolinux** [reply]
I mainly like Flatpak apps from Flathub for the ease of installing and updating proprietary apps like Chrome and Zoom etc.
I would probably prefer the Firefox repo if I needed the latest Firefox version.

**Comment by 03tekno** [comment]
sudo install -d -m 0755 /etc/apt/keyrings
wget -q https://packages.mozilla.org/apt/repo-signing-key.gpg -O- | sudo tee /etc/apt/keyrings/packages.mozilla.org.asc > /dev/null
echo "deb [signed-by=/etc/apt/keyrings/packages.mozilla.org.asc] https://packages.mozilla.org/apt mozilla main" | sudo tee -a /etc/apt/sources.list.d/mozilla.list > /dev/null
echo '
Package: *
Pin: origin packages.mozilla.org
Pin-Priority: 1000
' | sudo tee /etc/apt/preferences.d/mozilla
sudo apt update && sudo apt install firefox -y
